db/db_iter.cc

```diff
@@ -930,6 +965,16 @@ void DBIter::FindPrevUserKey() {
   }
 }
 
+bool DBIter::TooManyInternalKeysSkipped() {
+  if ((max_skippable_internal_keys_ > 0) &&
+      (num_internal_keys_skipped_ > max_skippable_internal_keys_)) {
+    valid_ = false;
+    status_ = Status::Incomplete("Too many internal keys skipped.");
+    return true;
+  }
+  return false;
+}
+
```

Comment:
maybe if would be better to move `num_internal_keys_skipped_++;` before line 969.
Then you can do this instead an if...else...
```
if (TooManyInternalKeysSkipped()) {
    return;
}
```